Assets/Scripts/Dialogue/InteractiveNPC.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class InteractiveNPC : MonoBehaviour
+{
+    private bool IsDisplayingDialogue;
+    private bool IsPlayerInRange;
+    private float DistanceBetweenPlayer;
+    public float MinDistance = 3.5f;
+    public DialogueManager DialogueManager;
+    public Conversation Conversation;
+
+    void Start()
+    {
+        IsDisplayingDialogue = false;
+        IsPlayerInRange = false;
+        DistanceBetweenPlayer = 0f;
+    }
+
+    void Update()
+    {
+        CheckIfInRange();
+
+        if(IsDisplayingDialogue & Input.GetButtonDown("Action Command"))
+        {
+            IsDisplayingDialogue = DialogueManager.AdvanceConversation();
+        }
+        else if (Input.GetButtonDown("Action Command") && IsPlayerInRange)
+        {
+            IsDisplayingDialogue = !IsDisplayingDialogue;
```

Comment:
This should be set to `true` I think. (even if you expect `IsDisplayingDialogue` to be false here, if the result you want is for it to become true, don't invert it! Instead, set it explicitly to true)